src/net.md

```diff
@@ -876,6 +877,61 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-extract-mediawiki-links]: #ex-extract-mediawiki-links
+<a name="ex-extract-mediawiki-links"/>
+## Extract all unique links from a MediaWiki markup
+
+[![reqwest-badge]][reqwest] [![regex-badge]][regex] [![cat-net-badge]][cat-net]
+
+Pull the source of a MediaWiki page using [`reqwest::get`] and then
+look for all entries of internal and external links with
+[`Regex::captures_iter`].
+
+MediaWiki syntax is described [here][MediaWiki link syntax].
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+extern crate regex;
+
+use std::collections::HashSet;
+use std::io::Read;
+use regex::Regex;
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Reqwest(reqwest::Error);
+#         Regex(regex::Error);
+#     }
+# }
+# 
+fn run() -> Result<()> {
+    let mut content = String::new();
+    reqwest::get(
+        "https://en.wikipedia.org/w/index.php?title=Rust_(programming_language)&action=raw",
+    )?
+        .read_to_string(&mut content)?;
+
+    let mut links = Regex::new(r#"\[\[(?P<link>[^\[\]|]*)[^\[\]]*\]\]"#)?
```

Comment:
sorry I meant `match` returning `Cow<str>` (damn autocorrect on mobile)
you would write something along the lines of:

```rust
match (c["internal"], c["external"]) {
        (Some(val), None) => Cow::from(val.to_lowercase()),
        (None, Some(val)) => Cow::from(val),
        _ => unreachable!(),
    }
```
if tje unreachable would be triggered then there is a fundamental error with the regex definition so we want to fail noisily and fix our code and not try to recocer